src/openfl/text/TextField.hx

```diff
@@ -3059,6 +3063,55 @@ class TextField extends InteractiveObject
 			scrollV -= event.delta;
 		}
 	}
+	
+	@:noCompletion private function this_onDoubleClick(event:MouseEvent):Void
+	{
+		if (__textEngine.selectable) {
+			__updateLayout();
+			
+			var delimiters:Array<String> = ['\n', '.', '!', '?', ',', ' ', ';', ':', '(', ')', '-', '_', '/'];
+			
+			var txtStr:String = __text;
+			var leftPos:Int = -1;
+			var rightPos:Int = txtStr.length;
+			var pos:Int = 0;
+			var startPos:Int = Std.int( Math.max(__caretIndex, 1) );
+			if (txtStr.length > 0 && __caretIndex >= 0 && rightPos >= __caretIndex)
+			{
+				for(c in delimiters)
+				{
+					pos = txtStr.lastIndexOf(c, startPos - 1);
+					if (pos > leftPos) leftPos = pos + 1;
+
+					pos = txtStr.indexOf(c, startPos);
+					if (pos < rightPos && pos != -1) rightPos = pos;
+				}
+				
+				if (leftPos != rightPos)
+				{
+					__selectionIndex = leftPos;
```

Comment:
`setSelection()` will adjust the scrollV as well which you need, instead of manually setting the indices

Reply:
Sure thing.